test/e2e/e2e_helpers_test.go

```diff
@@ -424,6 +423,65 @@ func sortNamespacedNames(nns []types.NamespacedName) {
 }
 
 func testDataPath(description ginkgo.GinkgoTestDescription) string {
-       path := filepath.Join(description.ComponentTexts...)
-       return strings.ReplaceAll(path, " ", "_")
+	path := filepath.Join(description.ComponentTexts...)
+	return strings.ReplaceAll(path, " ", "_")
+}
+
+func GetIstioObject(istio *istiov1beta1.Istio, namespace, name string) error {
+	return testEnv.Client.Get(context.TODO(), client.ObjectKey{
+		Namespace: namespace,
+		Name:      name},
+		istio)
+}
```

Comment:
I think this would be cleaner:

```suggestion
func GetIstioObject(namespace, name string) (*istiov1beta1.Istio, error) {
	var istio istiov1beta1.Istio
	err := testEnv.Client.Get(context.TODO(), client.ObjectKey{
		Namespace: namespace,
		Name:      name},
		&istio)
	return &istio, err
}
```

(I would probably use `istiov1beta1.Istio` as the result type, but the pointer might come in handy on the caller side, so either is fine)

Reply:
This method is called several times, and it seems problematic to create the istio instance inside the method, and then pass the address of that object outside the method.
It also seems wasteful to pass the entire object back each time this method is called.
Keeping this as-is

Reply:
This is just a nit so I'm fine with leaving this as-is, but just to be clear:
- you can "stack allocate" a value and then return a pointer to it. I'm using quotes because there is no stack allocation in go. The compiler can allocate a value on the stack if it can prove that a pointer to that value doesn't escape the function, but this is just an optimization and it mustn't change the meaning of your program. If it does, then that's a compiler bug. So you are free to use any pointer any way you like.
- readable code is preferable to fast code, especially since most of the time is spent waiting for changes in the cluster.